es/resource_elasticsearch_snapshot_lifecycle_policy_test.go

```diff
@@ -0,0 +1,166 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccElasticsearchSnapshotLifecyclePolicy,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckElasticsearchSnapshotLifecyclePolicyExists("elasticsearch_snapshot_lifecycle_policy.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccElasticsearchSnapshotLifecyclePolicy_importBasic(t *testing.T) {
+	provider := Provider().(*schema.Provider)
+	err := provider.Configure(&terraform.ResourceConfig{})
+	if err != nil {
+		t.Skipf("err: %s", err)
+	}
+	meta := provider.Meta()
+	var allowed bool
+	switch meta.(type) {
+	case *elastic5.Client, *elastic6.Client:
+		allowed = false
+	default:
+		allowed = true
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			if !allowed {
+				t.Skip("Snapshot lifecycles only supported on ES >= 7")
+			}
+		},
+		Providers:    testAccXPackProviders,
+		CheckDestroy: testCheckElasticsearchSnapshotLifecyclePolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccElasticsearchSnapshotLifecyclePolicy,
+			},
+			{
+				ResourceName:      "elasticsearch_snapshot_lifecycle_policy.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckElasticsearchSnapshotLifecyclePolicyExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No snapshot lifecycle policy ID is set")
+		}
+
+		meta := testAccXPackProvider.Meta()
+
+		var err error
+		switch client := meta.(type) {
+		case *elastic7.Client:
+			_, err = client.PerformRequest(context.TODO(), elastic7.PerformRequestOptions{Method: http.MethodGet, Path: "/_slm/" + name})
+		default:
+			err = errors.New("Snapshot Lifecycle Management is only supported by the elastic library >= v7!")
+		}
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testCheckElasticsearchSnapshotLifecyclePolicyDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "elasticsearch_snapshot_lifecycle_policy" {
+			continue
+		}
+
+		meta := testAccXPackProvider.Meta()
+
+		var err error
+		switch client := meta.(type) {
+		case *elastic7.Client:
+			_, err = client.XPackIlmGetLifecycle().Policy(rs.Primary.ID).Do(context.TODO())
+		default:
+			err = errors.New("Snapshot Lifecycle Management is only supported by the elastic library >= v7!")
+		}
+
+		if err != nil {
+			return nil // should be not found error
+		}
+
+		return fmt.Errorf("Snapshot lifecycle policy %q still exists", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+var testAccElasticsearchSnapshotLifecyclePolicy = `
+resource "elasticsearch_snapshot_lifecycle_policy" "test" {
+  name = "terraform-test"
+  body = <<EOF
+{
+  "schedule": "0 30 1 * * ?", 
+  "name": "<daily-snap-{now/d}>", 
+  "repository": "my_repository", 
```

Comment:
A check for this is failing the tests https://travis-ci.org/github/phillbaker/terraform-provider-elasticsearch/jobs/692706071#L1228

We can probably create the dependent repository using, for example:

https://github.com/phillbaker/terraform-provider-elasticsearch/blob/19c1ee3dd91c35e124069549452543b410d2dc8d/es/resource_elasticsearch_snapshot_repository_test.go#L111-L118